extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/ContextualEmitter.java

```diff
@@ -0,0 +1,53 @@
+package io.quarkus.smallrye.reactivemessaging.runtime;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+
+import io.smallrye.mutiny.Uni;
+import io.smallrye.reactive.messaging.EmitterType;
+
+/**
+ * Emitter implementation that plays better with context propagation, especially with the request scoped context.
+ * The message emit of this emitter makes sure that
+ * <p>
+ * <ul>
+ * <li>the message context is captured when active thread context plan is applied</li>
+ * <li>the message is emitted only after the context propagation ends the context switch, destroying the intermediate state for
+ * cleared contexts.</li>
+ * </ul>
+ *
+ * @param <T> the payload type
+ */
+public interface ContextualEmitter<T> extends EmitterType {
+
+    /**
+     * Send a payload
+     *
+     * @param payload the payload
+     * @return a Uni that completes when the message is sent
+     */
+    Uni<Void> send(T payload);
+
+    /**
+     * Send a payload and wait for the message to be sent
+     *
+     * @param payload the payload
+     */
+    void sendAndAwait(T payload);
```

Comment:
Don't we also have the `andForget` method?

Reply:
In the first iteration, I thought the `andForget` wouldn't make sense. But now it does. I'll add it.